src/modules/DistributionManager.sol

```diff
@@ -0,0 +1,535 @@
+        address _yieldToken,
+        address _yieldSource,
+        uint256 _cycleLength,
+        uint256 _yieldFixedSplitDivisor
+    ) internal {
+        if (_yieldToken == address(0)) revert ZeroAddress();
+        if (_yieldSource == address(0)) revert ZeroAddress();
+        if (_cycleLength == 0) revert InvalidCycleLength();
+        if (_yieldFixedSplitDivisor == 0) revert InvalidDivisor();
+
+        yieldToken = _yieldToken;
+        yieldSource = _yieldSource;
+        cycleLength = _cycleLength;
+        yieldFixedSplitDivisor = _yieldFixedSplitDivisor;
+        lastDistributionBlock = block.number;
+
+        _initializeOwner(msg.sender);
+    }
+
+    /// @inheritdoc IDistributionModule
+    function distributeYield() external override nonReentrant whenNotPaused {
+        (bool canDistribute,) = validateDistribution();
+        if (!canDistribute) revert DistributionNotResolved();
+
+        // Hook for token minting before distribution
+        _mintTokensBeforeDistribution();
+
+        // Collect yield
+        uint256 totalYield = _collectYield();
+        if (totalYield == 0) revert InsufficientYield();
+
+        // Calculate splits
+        (uint256 fixedAmount, uint256 votedAmount) = _calculateSplits(totalYield);
+
+        // Get distribution data
+        address[] memory activeRecipients = _getActiveRecipients();
+        if (activeRecipients.length == 0) revert NoRecipients();
+
+        uint256[] memory votes = _getVotingResults();
+
+        // Calculate distributions
+        uint256[] memory votedDistributions = _calculateVotedDistributions(votes, totalVotes, votedAmount);
+        uint256[] memory fixedDistributions = _calculateFixedDistributions(activeRecipients, fixedAmount);
+
+        // Execute distributions
+        _executeDistributions(activeRecipients, votedDistributions, fixedDistributions);
+
+        // Complete cycle
+        _completeCycleTransition();
+
+        // Record distribution
+        _recordDistribution(totalYield, totalVotes, activeRecipients, votedDistributions, fixedDistributions);
+
+        emit YieldDistributed(totalYield, totalVotes, activeRecipients, votedDistributions, fixedDistributions);
+    }
+
+    /// @inheritdoc IDistributionModule
+    function getCurrentDistributionState() external view override returns (DistributionState memory state) {
+        state.totalYield = _getAvailableYield();
+        (state.fixedAmount, state.votedAmount) = _calculateSplits(state.totalYield);
+        state.totalVotes = totalVotes;
+        state.lastDistributionBlock = lastDistributionBlock;
+        state.cycleNumber = cycleNumber;
+        state.recipients = recipients;
+        state.votedDistributions = new uint256[](recipients.length);
+        state.fixedDistributions = new uint256[](recipients.length);
+    }
+
+    /// @inheritdoc IDistributionModule
+    function validateDistribution() public view override returns (bool canDistribute, string memory reason) {
```

Comment:
rename this to isDistribution ready , add a call to the mock voting module to check that votes are above 0 

add a mock call to the token to see if there's >0 yield accured 

add a call to the mock recipient registry to see if there's > 0 participaints